### PR TITLE
Do not warn about enabled ABIEncoderV2 anymore.

### PR DESCRIPTION
**Pull request by chriseth**
Closes https://github.com/ethereum/solidity/issues/7679

### CODE REVIEW

**Comment by leonardoalt** [comment]
Needs rebase I guess

File: test/libsolidity/gasTests/abiv2.sol [hunk 1]
```diff
@@ -14,9 +14,9 @@ contract C {
 }
 // ----
 // creation:
-//   codeDepositCost: 1122800
-//   executionCost: 1167
-//   totalCost: 1123967
+//   codeDepositCost: 1120000
```
Comment (leonardoalt) [inline test/libsolidity/gasTests/abiv2.sol:17]: Why did this change? Metadata somehow?

File: libsolidity/ast/ExperimentalFeatures.h [hunk 2]
```diff
@@ -35,10 +36,11 @@ enum class ExperimentalFeature
 	TestOnlyAnalysis
 };
 
-static std::map<ExperimentalFeature, bool> const ExperimentalFeatureOnlyAnalysis =
+static std::set<ExperimentalFeature> const ExperimentalFeatureWithoutWarning =
 {
-	{ ExperimentalFeature::SMTChecker, true },
-	{ ExperimentalFeature::TestOnlyAnalysis, true },
+	ExperimentalFeature::ABIEncoderV2,
+	ExperimentalFeature::SMTChecker,
```
Comment (leonardoalt) [inline libsolidity/ast/ExperimentalFeatures.h:42]: Sure about this one? ;)

File: test/libsolidity/gasTests/abiv2.sol (hunk 1 above)
Comment (chriseth) [reply test/libsolidity/gasTests/abiv2.sol:17]: Yes, the cbor thing adds an "EXPERIMENTAL: true" string (or something similar).

**Comment by chriseth** [comment]
Updated.

File: test/libsolidity/gasTests/abiv2.sol (hunk 1 above)
Comment (leonardoalt) [reply test/libsolidity/gasTests/abiv2.sol:17]: Ok!

**Comment by leonardoalt** [comment]
@chriseth what exactly was updated?

**Comment by chriseth** [comment]
@leonardoalt the gas values

**Comment by chriseth** [comment]
Hopefully fixed the cmdline tests aswell.

**Review by leonardoalt — APPROVED** [review]

**Comment by chriseth** [comment]
Argh, forgot to check in two files...

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `b1396334f48847aca65ca3471cc92af3063cbef7`:
```
Error: Trailing whitespace found:
 test/cmdlineTests/abiencoderv2_no_warning/output:3:Function signatures: 

```
Please check that your changes are working as intended.

**Comment by chriseth** [comment]
Now depends on https://github.com/ethereum/solidity/pull/7772 (also being merged into 0.6.0)

File: libsolidity/ast/ExperimentalFeatures.h (hunk 2 above)
Comment (erak) [reply libsolidity/ast/ExperimentalFeatures.h:42]: Yeah, is this meant to go out of experimental with warning as well?

**Review by erak — COMMENTED** [review]
Not sure about `pragma experimental SMTChecker`, but the rest looks good!

File: libsolidity/ast/ExperimentalFeatures.h (hunk 2 above)
Comment (chriseth) [reply libsolidity/ast/ExperimentalFeatures.h:42]: No, it has always been....
Am I missing something here?

Comment (leonardoalt) [reply libsolidity/ast/ExperimentalFeatures.h:42]: @chriseth you're removing the experimental warning for the SMTChecker as well. Was that intended?

**Comment by erak** [comment]
Solc-js test is failing? Does this need to have https://github.com/ethereum/solidity/pull/7782 merged before, such that the new solc-js callback is in?

**Comment by leonardoalt** [comment]
Rebased

**Comment by leonardoalt** [comment]
@erak any idea why these are failing?

File: libsolidity/ast/ExperimentalFeatures.h (hunk 2 above)
Comment (chriseth) [reply libsolidity/ast/ExperimentalFeatures.h:42]: SMTChecker was inside this list before the change.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**